standard/PortalStatistics.lua

```diff
@@ -0,0 +1,1140 @@
+
+	local opponentNames = Array.map(Array.reverse(topEarningsList), function(opponent)
+				return config.opponentType == Opponent.team and opponent.name or opponent.id
+			end)
+
+	if Logic.readBool(config.yearBreakdown) then
+		return StatisticsPortal._buildChartData(config, yearSeriesData, opponentNames)
+	else
+		local chartData = {}
+		chartData[1] = {
+			name = 'Total Earnings',
+			type = 'bar',
+			data = StatisticsPortal._addArrays(yearSeriesData),
+		}
+
+		config.yAxis = {type = 'value', name = 'Earnings ($USD)'}
+		config.xAxis = {type = 'category', name = config.catLabel, data = opponentNames}
+		config.customLegend = config.customLegend or config.customInputs
+		return StatisticsPortal._drawChart(config, chartData)
+	end
+end
+
+
+--[[
+Section: Coverage Breakdown
+]] --
+
+
+function StatisticsPortal.coverageStatistics(args)
+
+	args = args or {}
+	args.alignSide = Logic.readBoolOrNil(args.alignSide)
+
+	local wrapper = mw.html.create('div')
+	local TournamentTable = wrapper:tag('div')
+	local matchTable = wrapper:tag('div')
+
+	if args.alignSide then
+		TournamentTable
+			:addClass('template-box')
+			:css('padding-right','2em')
+		matchTable
+			:addClass('template-box')
+			:css('padding-right','2em')
+	end
+
+	TournamentTable:node(StatisticsPortal.coverageTournamentTable(args))
+	matchTable:node(StatisticsPortal.coverageMatchTable(args))
+
+	return wrapper
+end
+
+
+function StatisticsPortal.coverageMatchTable(args)
+
+	args = args or {}
+	args.multiGame = Logic.readBoolOrNil(args.multiGame) or false
+	args.customGames = (type(args.customGames) == 'table' and args.customGames)
+		or (String.isNotEmpty(args.customGames) and Array.map(mw.text.split(args.customGames, ',', true), String.trim))
+		or GAMES
+
+	local matchTable = mw.html.create('table')
+		:addClass('wikitable wikitable-striped')
+
+	matchTable:tag('caption')
+		:wikitext(args.matchTableTitle or (Logic.readBool(args.alignSide) and '<br>' or ''))
+		:css('text-align','center')
+
+	local matchHeader = matchTable:tag('tr')
+
+	if Logic.readBool(args.multiGame) then
+		matchHeader:tag('th')
+			:wikitext('Game')
+	end
+
+	matchHeader
+		:tag('th'):wikitext(args.matchesTitle or 'Matches')
+		:tag('th'):wikitext(args.gamesTitle or 'Games')
+
+	if Logic.readBool(args.multiGame) then
+		for _, game in Table.iter.spairs(args.customGames) do
+			matchTable:node(StatisticsPortal._coverageMatchTableRow(args, {game = game, year = args.year}))
+		end
+	end
+
+	matchTable:node(StatisticsPortal._coverageMatchTableRow(args, {year = args.year}))
+
+	return matchTable
+end
+
+
+function StatisticsPortal._coverageMatchTableRow(args, parameters)
+
+	local resultsRow = mw.html.create('tr')
+	local tagtype = 'td'
+
+	if Logic.readBool(args.multiGame) and parameters.game == nil then
+		tagtype = 'th'
+		resultsRow:tag(tagtype)
+			:wikitext('Total')
+	elseif Logic.readBool(args.multiGame) and parameters.game ~= nil then
+		resultsRow:tag(tagtype)
+			:wikitext(parameters.game)
+	end
```

Comment:
extract to function since it is used several times

Reply:
But it sets the local tagtype variable, so I still need the first if condition